### PR TITLE
remake nvidia docker

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

make the Nvidia image lighter

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: dockers/nvidia/Dockerfile [hunk 1]
```diff
@@ -12,26 +12,68 @@
...
     if [ ! -z "$LIGHTNING_VERSION" ] ; then \
         rm -rf pytorch-lightning ; \
         wget https://github.com/PyTorchLightning/pytorch-lightning/archive/${LIGHTNING_VERSION}.zip --progress=bar:force:noscroll ; \
         unzip ${LIGHTNING_VERSION}.zip ; \
         mv pytorch-lightning-*/ pytorch-lightning ; \
         rm *.zip ; \
     fi && \
-    pip install ./pytorch-lightning["extra"] --no-cache-dir && \
+
+# Instalations
```
Comment (carmocca) [inline dockers/nvidia/Dockerfile:72]:
```suggestion
# Installations
```

**Review by carmocca — APPROVED** [review]

File: dockers/nvidia/Dockerfile [hunk 2]
```diff
@@ -12,26 +12,68 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM nvcr.io/nvidia/pytorch:21.02-py3
+FROM nvcr.io/nvidia/cuda:11.2.2-runtime-ubuntu20.04
```
Comment (SeanNaren) [inline dockers/nvidia/Dockerfile:15]: IIRC PyTorch suggested to stay at 11.1 due to some issues with 11.2, so we might want to as well?

**Review by SeanNaren — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

File: dockers/nvidia/Dockerfile [hunk 3]
```diff
@@ -12,26 +12,68 @@
...
 MAINTAINER PyTorchLightning <https://github.com/PyTorchLightning>
 
 ARG LIGHTNING_VERSION=""
 
-COPY ./ ./pytorch-lightning/
+SHELL ["/bin/bash", "-c"]
+# https://techoverflow.net/2019/05/18/how-to-fix-configuring-tzdata-interactive-input-when-building-docker-images/
+ENV \
+    DEBIAN_FRONTEND=noninteractive \
+    TZ=Europe/Prague \
```
Comment (tchaton) [inline dockers/nvidia/Dockerfile:25]: Why set Prague as TZ ?

Comment (tchaton) [reply dockers/nvidia/Dockerfile:25]: Do you build it locally on your computer ?

**Review by tchaton — APPROVED** [review]
LGTM !

File: dockers/nvidia/Dockerfile (hunk 3 above)
Comment (Borda) [reply dockers/nvidia/Dockerfile:25]: when ever you build you need to set a zone, otherwise it does not allow you got further...

**Pull request merged**